perfkitbenchmarker/providers/yandexcloud/yc_virtual_machine.py

```diff
@@ -0,0 +1,484 @@
+
+  def _NeedsToParseDescribeResponse(self):
+    """Returns whether the ID and IP addresses still need to be set."""
+    return not self.id or not self.internal_ip or not self.ip_address
+
+  @vm_util.Retry()
+  def _PostCreate(self):
+    """Get the instance's data."""
+    if self._NeedsToParseDescribeResponse():
+      getinstance_cmd = util.YcCommand(self, 'compute', 'instances',
+                                       'describe', self.name)
+      stdout, _, _ = getinstance_cmd.Issue()
+      response = json.loads(stdout)
+      self._ParseDescribeResponse(response)
+    if not self.image:
+      getdisk_cmd = util.YcCommand(
+          self, 'compute', 'disks', 'describe', self.name)
+      stdout, _, _ = getdisk_cmd.Issue()
+      response = json.loads(stdout)
+      self.image = response['source_image_id']
+      self.backfill_image = True
+    self.InstallPackages('git')
+
+  def _Delete(self):
+    """Delete a YC VM instance."""
+    delete_cmd = util.YcCommand(self, 'compute', 'instances', 'delete',
+                                self.name)
+    delete_cmd.Issue()
+
+  def _Exists(self):
+    """Returns true if the VM exists."""
+    getinstance_cmd = util.YcCommand(self, 'compute', 'instances',
+                                     'describe', self.name)
+    stdout, _, _ = getinstance_cmd.Issue(suppress_warning=True, raise_on_failure=False)
+    try:
+      response = json.loads(stdout)
+    except ValueError:
+      return False
+    try:
+      # The VM may exist before we can fully parse the describe response for the
+      # IP address or ID of the VM. For example, if the VM has a status of
+      # provisioning, we can't yet parse the IP address. If this is the case, we
+      # will continue to invoke the describe command in _PostCreate above.
+      # However, if we do have this information now, it's better to stash it and
+      # avoid invoking the describe command again.
+      self._ParseDescribeResponse(response)
+    except (KeyError, IndexError):
+      pass
+    return True
+
+  def ShouldDownloadPreprovisionedData(self, module_name, filename):
+    """Returns whether or not preprovisioned data is available."""
+    return False
+
+  def CreateScratchDisk(self, disk_spec):
+    """Create a VM's scratch disk.
+
+    Args:
+      disk_spec: virtual_machine.BaseDiskSpec object of the disk.
+    """
+    disks = []
+
+    for i in range(disk_spec.num_striped_disks):
+      name = '%s-data-%d-%d' % (self.name, len(self.scratch_disks), i)
+      data_disk = yc_disk.YcDisk(disk_spec, name, self.zone, self.folder_id)
+      data_disk.disk_number = self.remote_disk_counter + 1
+      self.remote_disk_counter += 1
+      disks.append(data_disk)
+
+    self._CreateScratchDiskFromDisks(disk_spec, disks)
+
+  def AddMetadata(self, **kwargs):
+    """Adds metadata to the VM and disk."""
+    if not kwargs:
+      return
+    cmd = util.YcCommand(self, 'compute', 'instances', 'add-metadata',
```

Comment:
I don't see add metadata as an option in https://cloud.yandex.com/docs/cli/command-tree/manage-services/compute

Reply:
Our documentation is partially outdated. Really yc has add-metadata option:

```
Manage virtual machine instances

Usage:
  yc compute instance <command>

Aliases:
  instances

Commands:
  get                    Show information about the specified virtual machine instance
  list                   List virtual machine instances
  create                 Create a virtual machine instance
  create-with-container  Create a virtual machine instance running Docker container
  update                 Update the specified virtual machine instance
  update-container       Update the specified virtual machine instance running Docker container
  add-metadata           Add or update metadata for the specified virtual machine instance
  remove-metadata        Remove keys from metadata for the specified virtual machine instance
  add-labels             Add labels to specified virtual machine instance
  remove-labels          Remove labels from specified virtual machine instance
  delete                 Delete the specified virtual machine instance
  get-serial-port-output Return the serial port output of the specified virtual machine instance
  stop                   Stop the specified virtual machine instance
  start                  Start the specified virtual machine instance
  restart                Restart the specified virtual machine instance
  attach-disk            Attach existing disk to the the specified virtual machine instance
  attach-new-disk        Attach new disk to the the specified virtual machine instance
  detach-disk            Detach disk from the the specified virtual machine instance
  add-one-to-one-nat     Add one-to-one NAT to the the specified network interface of virtual machine instance
  remove-one-to-one-nat  Remove one-to-one NAT to the the specified network interface of virtual machine instance
  list-operations        List operations for the specified instance
```